### PR TITLE
Fix decimals override (BSC balances issue)

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-####

## What changed (plus any additional context for devs)
Fixes the issue where we were using incorrect decimal numbers for some BSC assets.
This wasn't a BSC specific issue but we only saw it there bc it's the only chain I know where the token decimals doesn't match the mainnet token decimals.

The problem
This wasn't working at all in production:
https://github.com/rainbow-me/rainbow/blob/develop/src/utils/getTokenMetadata.ts#L11-L13

causing all the token metadata to be overwritten, including chain specific data.

I believe it's because we're running JS in strict mode and some things are different but I'm not gonna spend more time figuring out why.

TLDR: Use `omitFlatten` if possible or return a new object instead of using delete in the future to avoid surprises

## Screen recordings / screenshots
Before:
Screen Shot 2023-02-03 at 3 20 06 PM

After:
Screen Shot 2023-02-03 at 5 25 59 PM

## What to test
Make a testflight build and an android apk for this branch and confirm the balances for 0xtester.eth are showing correctly

### CODE REVIEW

**Review by skylarbarrera — APPROVED** [review]

**Review by estrattonbailey — APPROVED** [review]

**Review by BrodyHughes — APPROVED** [review]
Working as expected on iOS + Android

**Review by ibrahimtaveras00 — APPROVED** [review]
Checked both OS's, 1.8.9 (11) and release APK, QA Passed 👍🏽
